diesel/src/pg/types/array.rs

```diff
@@ -48,7 +49,7 @@ where
                 } else {
                     let (elem_bytes, new_bytes) = bytes.split_at(elem_size as usize);
                     bytes = new_bytes;
-                    T::from_sql(Some(elem_bytes))
+                    T::from_sql(Some(PgValue::with_oid(elem_bytes.as_ptr() as *mut u8, elem_bytes.len(), oid as u32)))
```

Comment:
Maybe we should make the `PgValue::with_oid` constructor just taking a `&[u8]` slice and do the constructing of that slice only once in `pg/connection/cursor.rs`?